src/components/TagForm/index.tsx

```diff
@@ -0,0 +1,54 @@
+'use client';
+
+import { TagSchemaType, tagSchema } from '@/lib/zod/tags';
+import { TextField, Button, Box } from '@mui/material';
+import { useForm } from 'react-hook-form';
+import { useRouter } from 'next/navigation';
+import { zodResolver } from '@hookform/resolvers/zod';
+import FormFieldError from '@/components/FormFieldError/FormFieldError';
+import { useMessage } from '../Providers/MessageProvider';
+import { post } from '@/lib/response/fetchUtil';
+
+export default function TagForm() {
+  const router = useRouter();
+  const { notify } = useMessage();
+
+  const {
+    register,
+    formState: { errors, isSubmitting },
+    handleSubmit,
+    reset,
+  } = useForm<TagSchemaType>({
+    resolver: zodResolver(tagSchema),
+  });
+
+  const submitForm = async (data: TagSchemaType) => {
+    data.name = data.name.replace(/\s{2,}/g, ' ').trim();
+    const responseJson = await post('/api/tag', data);
+    notify(responseJson);
+    reset();
+    router.refresh();
+  };
+
+  return (
+    <>
+      <Box sx={{ mt: 1, mb: 4 }}>
+        <form onSubmit={handleSubmit(submitForm)}>
+          <TextField label="Tag name" {...register('name')}></TextField>
```

Comment:
```suggestion
          <TextField label="Tag name" {...register('name')} inputProps={{ maxLength: 51 }}></TextField>
```

If you want, you can just have in frontend the user stop writing once the message error has appeared with the inputProps maxlenght attribute. Right now the use can keep typing despite the lenght error message. If you choose to do that, not having it as a magic number would be better though. Up to you.

Reply:
Good idea, I implemented this (and got rid of the magic number)